zerver/tornado/event_queue.py

```diff
@@ -231,6 +235,16 @@ def accepts_event(self, event: Mapping[str, Any]) -> bool:
             # events are sent only if user_settings_object is False,
             # otherwise only 'user_settings' event is sent.
             return False
+        if event["type"] == "user_group":
+            if event["op"] == "remove":
+                # 'user_group/remove' events are only sent if the client
+                # cannot filter out deactivated groups by themselves.
+                return not self.include_deactivated_groups
+            if event["op"] == "update" and "deactivated" in event["data"]:
+                # 'update' events for group deactivation are only sent to
+                # clients who can filter out deactivated groups by themselves.
+                # Other clients receive 'remove' event.
+                return self.include_deactivated_groups
```

Comment:
This makes sure that `update` event is sent to modern clients which can handle deactivated groups and `remove` to old clients.